clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesV2CachingAgent.java

```diff
@@ -138,7 +138,12 @@ protected CacheResult buildCacheResult(Map<KubernetesKind, List<KubernetesManife
         .peek(m -> RegistryUtils.removeSensitiveKeys(propertyRegistry, accountName, m))
         .map(rs -> {
           try {
-            return KubernetesCacheDataConverter.convertAsResource(accountName, rs, relationships.get(rs));
+            CacheData cacheData = KubernetesCacheDataConverter.convertAsResource(accountName, rs, relationships.get(rs));
+            if(credentials.getOnlySpinnakerManaged() && cacheData.getAttributes().get("application").equals("")) {
```

Comment:
Missing a space before `(`.

Also, I would use the apache `StringUtils.isNotEmpty(cacheData.getAttributes().get("application")` instead, it checks for `null` as well.

Reply:
Also, this might be cleaner to put into a `filter` block following this `map`.

Reply:
Last comment -- you need a similar guard here: https://github.com/spinnaker/clouddriver/pull/3014/files#diff-c4d66e4481b7757543a8868afa363ef8R163